patientsearch/src/js/components/TimeoutModal.js

```diff
@@ -53,14 +53,22 @@ export default function TimeoutModal() {
         let tokenData = null;
         try {
           tokenData = JSON.parse(response);
+          let accessTokenExpiresIn = parseFloat(tokenData["access_expires_in"]);
+          let refreshTokenExpiresIn = parseFloat(tokenData["refresh_expires_in"]);
+          let refreshTokenOnVentilator = refreshTokenExpiresIn < accessTokenExpiresIn;
           //in seconds
-          expiresIn = tokenData["expires_in"];
-
-          if (!tokenData["valid"] || expiresIn <= 1) {
+          //1. check if refresh token will expire before access token first
+          //2. check if access token will expire
+          expiresIn = refreshTokenOnVentilator ? refreshTokenExpiresIn : accessTokenExpiresIn;
+          let tokenAboutToExpire = Math.floor(expiresIn) >= 1 && Math.floor(expiresIn) <= 60;
+          //flag for whether to prompt the user to refresh session, i.e. request another access token
+          refresh = tokenAboutToExpire && !refreshTokenOnVentilator ? true: false;
+
+          if (refreshTokenOnVentilator && (!tokenData["valid"] || expiresIn <= 1)) {
             handleLogout();
             return;
           }
```

Comment:
Sorry! Realized it was more nuanced that I thought- we should clear the local session if the (Access) token has expired.
```suggestion
          if (!tokenData["valid"] || expiresIn <= 1) {
            if (refreshTokenOnVentilator){
              handleLogout();
            } else {
              reLoad();
            }
            return;
          }
```

Reply:
thank you @ivan-c I did just fix it